docs/x509/reference.rst

```diff
@@ -1119,6 +1119,21 @@ X.509 Extensions
 
         The binary value of the identifier.
 
+    .. classmethod:: from_public_key(public_key)
+
+        .. versionadded:: 1.0
+
+        Creates a new SubjectKeyIdentifier instance using the public key
+        provided to generate the appropriate digest. This should be the public
+        key that is in the certificate.
```

Comment:
That is correct, although it's the only option I've ever seen used. I'll add some more prose to explain that it's the SHA1 of the subjectPublicKey BIT STRING and link the RFC section